app/server/me_api/credit_transfer.py

```diff
@@ -229,19 +234,20 @@ def post(self):
             }
             return make_response(jsonify(response_object)), 401
 
-        if not my_transfer_account_id:
-            response_object = {
-                'message': 'You must provide your Transfer Account ID',
-            }
-            return make_response(jsonify(response_object)), 400
+        if not my_transfer_account:
+            if not my_transfer_account_id:
+                response_object = {
+                    'message': 'You must provide your Transfer Account ID',
+                }
+                return make_response(jsonify(response_object)), 400
 
-        my_transfer_account = TransferAccount.query.get(my_transfer_account_id)
+            my_transfer_account = TransferAccount.query.get(my_transfer_account_id)
```

Comment:
I might be missing something, but can a user just enter an arbitrary `my_transfer_account_id` here?

Reply:
+1

Reply:
just reviewed. It's definitely a very big problem. I've put in a quick fix and let any clients where we can't confirm whether the bug has been exploited or not know.

Reply:
We should audit this code properly - it's a bit of a mix of modalities and needs a good refactor i think